src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs

```diff
@@ -165,17 +171,12 @@ public override BoundNode VisitITuplePattern(BoundITuplePattern node)
                         // for property part
                         if (!rp.Properties.IsDefault)
                         {
-                            foreach (BoundSubpattern subpattern in rp.Properties)
+                            foreach (BoundPropertySubpattern subpattern in rp.Properties)
                             {
-                                if (subpattern.Symbols.IsEmpty)
-                                    continue;
-
-                                // Obtain the last symbol's slot which is actually the one being matched.
-                                Symbol last = subpattern.Symbols.Last();
-                                // PROTOTYPE(extended-property-patterns) For a chain of members, the input type is no longer `inputType`
-                                if (last.ContainingType.Equals(inputType, TypeCompareKind.AllIgnoreOptions))
+                                if (subpattern.Member is { Symbol: Symbol symbol, Receiver: var receiver } member &&
```

Comment:
>var [](http://example.com/codeflow?start=92&length=3)

I didn't understand how the nullability analysis of the chain of accesses falls out  from this implementation. I would have expected we'd analyze the nullability of the receiver first (the chain of receivers is not null) then learn that the current member/symbol is also not null.

nit: please use an explicit type here for readability

Reply:
Me neither. I'll have to investigate to see where the state is produced.

>nit: please use an explicit type here for readability

We don't want the null check here. Filed a suggestion to address this: https://github.com/dotnet/csharplang/issues/4724

Reply:
Let's keep a PROTOTYPE marker here then. Thanks